src/mockoptimizer.jl

```diff
@@ -216,21 +152,166 @@ mutable struct MockLinQuadOptimizer <: LQOI.LinQuadOptimizer
+    if !isempty(from)
+        out = from[1]
+        shift!(from)
+        return out
+    else
+        if warn
+            Compat.@warn("No data in the input vector, returning default.")
+        end
+        return default
+    end
+end
+
+"""
+    fakesolve(instance::MockLinQuadOptimizer)
+
+Set solutions upon solve calls.
+Data held in `MockLinQuadOptimizer` stored data is passed to the low-level emulator `MockLinQuadModel`.
+"""
+function fakesolve(instance::MockLinQuadOptimizer)
+
+    instance.inner.termination_status = unload(instance.termination_status_stored, instance.inner.termination_status)
+    instance.inner.primal_status = unload(instance.primal_status_stored, instance.inner.primal_status)
+    instance.inner.dual_status = unload(instance.dual_status_stored, instance.inner.dual_status)
+
+    instance.inner.variable_primal_solution = unload(instance.variable_primal_solution_stored, instance.inner.variable_primal_solution)
+    instance.inner.variable_dual_solution = unload(instance.variable_dual_solution_stored, instance.inner.variable_dual_solution)
+
+    instance.inner.constraint_primal_solution = unload(instance.constraint_primal_solution_stored, instance.inner.constraint_primal_solution)
+    instance.inner.constraint_dual_solution = unload(instance.constraint_dual_solution_stored, instance.inner.constraint_dual_solution)
+
+    instance.inner.quadratic_primal_solution = unload(instance.quadratic_primal_solution_stored, instance.inner.quadratic_primal_solution)
+    instance.inner.quadratic_dual_solution = unload(instance.quadratic_dual_solution_stored, instance.inner.quadratic_dual_solution)
+
+    nothing
+end
+
+function set_variable_primal_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.variable_primal_solution_stored, input)
+end
+function set_variable_dual_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.variable_dual_solution_stored, input)
+end
+function set_constraint_primal_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.constraint_primal_solution_stored, input)
+end
+function set_constraint_dual_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.constraint_dual_solution_stored, input)
+end
+function set_quadratic_dual_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.quadratic_dual_solution_stored, input)
+end
+function set_quadratic_primal_solution!(instance::MockLinQuadOptimizer, input)
+    push!(instance.quadratic_primal_solution_stored, input)
+end
+function set_termination_status!(instance::MockLinQuadOptimizer, input)
+    push!(instance.termination_status_stored, input)
+end
+function set_primal_status!(instance::MockLinQuadOptimizer, input)
+    push!(instance.primal_status_stored, input)
+end
+function set_dual_status!(instance::MockLinQuadOptimizer, input)
+    push!(instance.dual_status_stored, input)
+end
+
+function set_solution!(instance::MockLinQuadOptimizer;
+    variable_primal = nothing,
```

Comment:
These can be sensible defaults, e.g., `variable_primal = Float64[]`, then you can skip the `Nothing` checks.